sdk/src/job_runner.rs

```diff
@@ -7,38 +7,105 @@ use std::future::Future;
 use std::pin::Pin;
 use tangle_subxt::tangle_testnet_runtime::api;
 use tangle_subxt::tangle_testnet_runtime::api::runtime_types::tangle_primitives::services;
-use tangle_subxt::tangle_testnet_runtime::api::runtime_types::tangle_primitives::services::PriceTargets;
+use tangle_subxt::tangle_testnet_runtime::api::runtime_types::tangle_primitives::services::PriceTargets as TangleSubxtPriceTargets;
 
-pub struct MultiJobRunner<'a> {
-    pub(crate) enqueued_job_runners: EnqueuedJobRunners<'a>,
-    pub(crate) env: Option<GadgetConfiguration<parking_lot::RawRwLock>>,
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("No jobs registered. Make sure to add a job with `MultiJobRunner::add_job`")]
+    NoJobs,
+    #[error("Job already initialized")]
+    AlreadyInitialized,
+
+    #[error(transparent)]
+    Recv(#[from] tokio::sync::oneshot::error::RecvError),
 }
 
-pub type EnqueuedJobRunners<'a> = Vec<
-    Pin<
-        Box<
-            dyn ScopedFuture<
-                'a,
-                Output = Option<tokio::sync::oneshot::Receiver<Result<(), crate::Error>>>,
-            >,
-        >,
-    >,
->;
+/// Wrapper for `tangle_subxt`'s [`PriceTargets`]
+///
+/// This provides a [`Default`] impl for a zeroed-out [`PriceTargets`].
+///
+/// [`PriceTargets`]: tangle_subxt::tangle_testnet_runtime::api::runtime_types::tangle_primitives::services::PriceTargets
+pub struct PriceTargets(TangleSubxtPriceTargets);
```

Comment:
We can provide a default implementation in subxt and remove this wrapper. But it is good for now.